documentation/building/linux-instructions.md

```diff
@@ -1,90 +1,200 @@
 
-    ~$ sudo apt-get install make wget cmake llvm-3.9 clang-3.9 lldb-3.9 liblldb-3.9-dev gettext
+    sudo apt-get install cmake clang-3.9 gdb gettext git libicu-dev lldb-3.9 liblldb-3.9-dev libunwind8 llvm-3.9 make python python-lldb-3.9 tar zip
 
 The lldb 3.9 package needs a lib file symbolic link fixed:
 
     cd /usr/lib/llvm-3.9/lib
     sudo ln -s ../../x86_64-linux-gnu/liblldb-3.9.so.1 liblldb-3.9.so.1
 
-You now have all the required components.
+See the section below on how to clone, build and test the repo.
+
+#### Ubuntu 17.10 ####
+
+Add the additional package sources:
+
+    sudo apt-get update
+    sudo apt-get install wget
+    echo "deb http://llvm.org/apt/artful/ llvm-toolchain-artful main" | sudo tee /etc/apt/sources.list.d/llvm.list
+    echo "deb http://llvm.org/apt/artful/ llvm-toolchain-artful-3.9 main" | sudo tee -a /etc/apt/sources.list.d/llvm.list
+    wget -O - http://llvm.org/apt/llvm-snapshot.gpg.key | sudo apt-key add -
+    sudo apt-get update
 
-#### Alpine (3.6) ####
+Then install the required packages:
 
+    sudo apt-get install cmake clang-3.9 gdb gettext git libicu-dev lldb-3.9 liblldb-3.9-dev libunwind8 llvm-3.9 make python python-lldb-3.9 tar zip
 
-```sh
-~$ sudo apk add clang clang-dev cmake coreutils gcc gettext-dev linux-headers llvm lldb-dev make python
+#### Ubuntu 18.04 ####
 
-```
+Install the required packages:
 
-#### CentOS (6, 7) ####
+    sudo apt-get update
+    sudo apt-get install cmake clang-3.9 curl gdb gettext git libicu-dev libssl1.0-dev lldb-3.9 liblldb-3.9-dev libunwind8 llvm-3.9 make python-lldb-3.9 tar wget zip 
 
-llvm, clang and lldb 3.9 will have to be built for this distro ([TBD] add instructions).
+See the section below on how to clone, build and test the repo.
 
-```sh
-~$ sudo yum install centos-release-SCL epel-release
-~$ sudo yum install which wget make cmake cmake3 gcc gcc-c++ python27 python-argparse python-devel
+#### Alpine 3.6 ####
 
-```
+Currently there is no lldb that works on Alpine.
 
-#### Debian (8.2) ####
+Issue https://github.com/dotnet/diagnostics/issues/73
+
+#### CentOS 6 ####
 
 [TBD]
 
-#### Fedora (23, 24, 27, 28) ####
+#### CentOS 7 ####
+
+llvm, clang and lldb 3.9 will have to be built for this distro.
+
+First the prerequisites:
+
+    sudo yum install centos-release-SCL epel-release
+    sudo yum install cmake cmake3 gcc gcc-c++ gdb git libicu libunwind make python27 tar wget which zip
+
+Now build and install llvm/lldb 3.9 using the script provided here: [build-install-lldb.sh](../lldb/centos7/build-install-lldb.sh).
+
+WARNING: this script installs llvm and lldb and may overwrite any previously installed versions.
+
+    cd $HOME
+    git clone https://github.com/dotnet/diagnostics.git
+    $HOME/diagnostics/documentation/lldb/centos7/build-install-lldb.sh
+
+This will take some time to complete, but after it is finished all the required components will be built and installed. To build and test the repo:
+
+    cd diagnostics
```

Comment:
All the other distros deferred to the build/test instructions below, but this one inlined it. Unless there was a reason I missed lets normalize.